charts/emissary-ingress/values.yaml.in

```diff
@@ -353,7 +353,10 @@ resolvers: # +doc-gen:break
 # for more info on the available options.
 #
 # Note: The Module can only be named ambassador. There can only be one Module
-# installed per-namespace.
+# object installed per-namespace. To create multiple ambassador Modules in the
```

Comment:
@AliceProxy - What are your thoughts on this wording? I have never tested this so I assume it works but annotation support is discouraged and not actively maintained. 

I feel like we should add a caveat here that although this works that it is discouraged to use annotations because support for them is very limited and its likely they will go away.  

Also, I would think it would be a better recommendation to suggest users to isolate each instance of Emissary-ingress in its own namespace with separate Ambassador_IDs.

Reply:
FWIW, mind if I add my use-case?

We have services where some components require internal network traffic and some require public IPs, so we have one Emissary behind an internal load balancer and another Emissary behind an internet-facing load balancer. For us it makes little sense to separate those into different namespaces.

If annotation-based Modules are deprecated it would be great if it would be possible to create more than one Module in the same namespace instead.

Reply:
@eevdev - Thanks for sharing the use case. It makes sense.

I think the key thing for the Module is just making sure for each Emissary-ingress instance there is only a single Module found. Off the top of my head, I don't believe there are any limitations to supporting multiple in a namespace. I would be happy to accept a PR to support multiple Modules in a namespace.

As for the annotations, these are very legacy code that across the board should be removed but have been kept around for folks migrating from older versions and also just the effort to actually remove them so even if it does work as you mentioned in the comment. I just hesitate to add it as a recommendation due to the fact that they are legacy and at some point will go away. Where as architecting your solution with multiple namespaces is stable pattern that is fairly core to K8s.

Reply:
@LanceEa The limitation for why there cannot be multiple in a single namespace is that it is still a requirement that Modules must be named `ambassador`, so you can't have two resources of the same type with the same name.namespace. The annotations are a workaround of that limitation but I agree that we should not encourage further use of annotation based config. For now I say let's leave this as-is but we should probably just update the Module handling code to not care about the name and just handle multiple Modules not separated by ambassador ID as an error in the same way we do with the AuthService and RatelimitService resources. IMO let's just go ahead and land this, then if/when we make that change we can come back and remove mention of the annotations here.

Reply:
:) oh yeah, the whole namespace/name limitation. Thanks for input, I'm cool with landing as-is. Thanks @eevdev